n3fit/src/evolven3fit/evolven3fit_API.py

```diff
@@ -0,0 +1,209 @@
+from . import utils
+from validphys.loader import Loader
+import os
+from ekomark import apply
+from eko import basis_rotation as br
+from eko import run_dglap
+from eko import output
+import logging
+
+logger = logging.getLogger(__name__) 
+
+def evolve_fit(conf_folder, op_card_dict, t_card_dict ,eko_path=None, dump_eko=None):
+    """
+    Evolves all the fitted replica in conf_folder/nnfit 
+
+    Parameters
+    ----------
+
+        conf_folder: str or pathlib.Path
+            path to the folder containing the fit
+        op_card_dict: dict
+            user settings for the op_card
+        t_card_dict: dict
+            user settings for the t_card
+        eko_path: str or pathlib.Path
+            path where the eko is stored (if None the eko will be
+            recomputed)
+        dump_eko: str or pathlib.Path
+            path where the eko is dumped (if None the eko won't be
+            stored)
+    """
+    log_file = pathlib.Path(conf_folder) / f"evolven3fit.log"
+    if log_file.exists():
+        log_file.unlink()
+    log_file = logging.FileHandler(log_file)
+    log_file.setLevel(logging.INFO)
+    log_file.setFormatter(
+            logging.Formatter("%(asctime)s %(name)s/%(levelname)s: %(message)s")
+        )
+    for logger_ in (logger, *[logging.getLogger("eko")]):
+        logger_.handlers = []
+        logger_.setLevel(logging.INFO)
+        logger_.addHandler(log_file)
+    usr_path = pathlib.Path(conf_folder)
+    initial_PDFs_dict = load_fit(usr_path)
+    theory, op = construct_eko_cards(usr_path, op_card_dict, t_card_dict)
+    if eko_path is not None:
+        eko_path = pathlib.Path(eko_path)
+        eko_op = output.Output.load_tar(eko_path) 
+    else:
+        eko_op = construct_eko_for_fit(theory, op, dump_eko)
+    info = gen_info.create_info_file(theory, op, 1, info_update={})  
+    info["NumMembers"] = "REPLACE_NREP"
+    info["ErrorType"] = "replicas"
+    dump_info_file(usr_path, info)
+    utils.fix_info_path(usr_path)
+    for replica in initial_PDFs_dict.keys():
+        evolved_block = evolve_exportgrid(initial_PDFs_dict[replica], eko_op, theory, op)
+        dump_evolved_replica(
+            evolved_block, usr_path, int(replica.removeprefix("replica_"))
+        )
+        # fixing_replica_path
+        utils.fix_replica_path(usr_path, int(replica.removeprefix("replica_")))
+    # remove folder
+    os.rmdir(usr_path / "nnfit" / usr_path.stem)
+
+
+def load_fit(usr_path):
+    """
+    Loads all the replica pdfs at fitting scale in usr_path and return the exportgrids
+
+    Parameters
+    ----------
+
+        usr_path: pathlib.Path
+            path to the folder containing the fit  
+    
+    Returns
+    -------
+
+            : dict
+            exportgrids info
+    """
+    nnfitpath = usr_path / "nnfit"
+    replica_list = []
+    for subdir, dirs, files in os.walk(nnfitpath):
+        for dir in dirs:
+            replica_list.append(dir)
+    replica_list.remove("input")
```

Comment:
I'm not sure I understand what the content of `replica_list` should be, but I'm guessing a `glob` rule would work better. Something like `nnfitpath.glob("replica_*")` will give you an iterator for all `replica_X` folders which you can iterate over and with exactly the content you want (no need for try-except later)